storage/Devices/PartitionImpl.cc

```diff
@@ -242,8 +243,26 @@ namespace storage
 
 	if (get_type() != EXTENDED)
 	{
-	    // TODO look at id
-	    cmd_line += "ext2 ";
+	    switch (get_id())
+	    {
+		case ID_SWAP:
+		    cmd_line += "linux-swap ";
+		    break;
+
+		case ID_GPT_BOOT:
+		case ID_DOS16:
```

Comment:
Copied from the old libstorage. AFAIS parted also supports fat16 and ntfs so I can add those.